jme3-effects/src/main/java/com/jme3/post/filters/ContrastAdjustmentFilter.java

```diff
@@ -0,0 +1,402 @@
+     *
+     * @param exponent the desired exponent (default=1)
+     * @return this filter instance, for chaining
+     */
+    public ContrastAdjustmentFilter setBlueExponent(float exponent) {
+        this.blueChannelExponent = exponent;
+        if (material != null) {
+            material.setFloat("blueChannelExponent", blueChannelExponent);
+        }
+        return this;
+    }
+
+    /**
+     * Retrieves the red channel exponent.
+     * Default value = 1.0f
+     *
+     * @return the red channel exponent.
+     */
+    public float getRedChannelExponent() {
+        return redChannelExponent;
+    }
+
+    /**
+     * Retrieves the green channel exponent.
+     * Default value = 1.0f.
+     *
+     * @return the green channel exponent.
+     */
+    public float getGreenChannelExponent() {
+        return greenChannelExponent;
+    }
+
+    /**
+     * Retrieves the blue channel exponent.
+     * Default value = 1.0f
+     *
+     * @return the blue channel exponent.
+     */
+    public float getBlueChannelExponent() {
+        return blueChannelExponent;
+    }
+
+    /**
+     * Sets the input range for all color channels. Before applying the power law, the input levels get
+     * normalized so that lowerLimit becomes 0 and upperLimit becomes 1.
+     *
+     * @param lowerLimit the desired lower limit (default=0)
+     * @param upperLimit the desired upper limit (default=1)
+     * @return this filter instance for a chain call.
+     */
+    public ContrastAdjustmentFilter setInputRange(float lowerLimit, float upperLimit) {
+        setLowerLimit(lowerLimit);
+        setUpperLimit(upperLimit);
+        return this;
+    }
+
+    /**
+     * Sets the upper limit of the input range.
+     *
+     * @param level the input level that should be normalized to 1 (default=1)
+     * @return this filter instance, for chaining
+     */
+    public ContrastAdjustmentFilter setUpperLimit(float level) {
+        this.upperLimit = level;
+        if (material != null) {
+            material.setFloat("upperLimit", upperLimit);
+        }
+        return this;
+    }
+
+    /**
+     * Sets the lower limit of the input range.
+     *
+     * @param level the input level that should be normalized to 0 (default=0)
+     * @return this filter instance, for chaining
+     */
+    public ContrastAdjustmentFilter setLowerLimit(float level) {
+        this.lowerLimit = level;
+        if (material != null) {
+            material.setFloat("lowerLimit", lowerLimit);
+        }
+        return this;
+    }
+
+    /**
+     * Returns the lower limit of the input range.
+     * Default value = 0.0.
+     *
+     * @return the lower limit
+     */
+    public float getInputRangeLowerLimit() {
+        return lowerLimit;
+    }
+
+    /**
+     * Returns the upper limit of the input range.
+     * Default value = 1.0.
+     *
+     * @return the upper limit
+     */
+    public float getInputRangeUpperLimit() {
```

Comment:
Could you please standardize the getter and setter methods of the variables? could you change to getUpperLimit ?

Reply:
Done.